### PR TITLE
Allow local packages in requirements.txt dependency list.

### PR DESCRIPTION
**Pull request by robertwb**
We pull these out and install them with the extra local packages infrastructure.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @chamikaramj

**Comment by AnandInguva** [comment]
How is this different from using `--extra_package`?

**Comment by robertwb** [comment]
Because the user can place these directly inside their requirements file rather than listing them individually. This is particularly valuable for cross-language, or where one wants to keep the construction-time and execution-time environments in sync via a single file.

**Comment by robertwb** [comment]
Failure due to #21104

**Comment by robertwb** [comment]
Run Python PreCommit

**Comment by robertwb** [comment]
Run Python_PVR_Flink PreCommit

**Comment by robertwb** [comment]
Good call. Updated the docs.

File: CHANGES.md [hunk 1]
```diff
@@ -50,6 +50,15 @@
 * ([#X](https://github.com/apache/beam/issues/X)).
 -->
 
+# [2.44.0] - Unreleased
+
+## New Features / Improvements
```
Comment (chamikaramj) [inline CHANGES.md:55]: Can we also add a unit test for this feature ?

**Comment by AnandInguva** [comment]
cc : @tvalentyn

File: CHANGES.md (hunk 1 above)
Comment (robertwb) [reply CHANGES.md:55]: Done.

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Pull request merged**